src/Service/ProducerSideServiceImpl.cpp

```diff
@@ -356,16 +358,32 @@ void ProducerSideServiceImpl::ReceiveEventsThread(
     uint64_t producer_id, bool* all_events_sent_received) {
   orbit_base::SetCurrentThreadName("PSSI::RcvEvents");
 
-  orbit_grpc_protos::ReceiveCommandsAndSendEventsRequest request;
-  while (stream->Read(&request)) {
+  // Read the ReceiveCommandsAndSendEventsRequest on a protobuf Arena, which prevents memory
```

Comment:
**Create** the ReceiveCommandsAndSendEventsRequest on a ...

Reply:
Done.